File: etl/models/motivo_cassacao.py
from sqlalchemy import Column, BigInteger, String, ForeignKey, PrimaryKeyConstraint
from .base import Base

class MotivoCassacao(Base):
    __tablename__ = "motivo_cassacao"

    sq_candidato = Column(BigInteger, ForeignKey("candidatura.sq_candidato"), nullable=False)
    ds_tp_motivo = Column(String)
    ds_motivo = Column(String, nullable=False)

    __table_args__ = (
        PrimaryKeyConstraint("sq_candidato", "ds_motivo"),
    )

File: src/backend/VoteBem/Entities/MotivoCassacao.cs
namespace VoteBem.Entities
{
    public class MotivoCassacao
    {
        public long SqCandidato { get; private set; }
        public string? DsTpMotivo { get; private set; }
        public string DsMotivo { get; private set; } = null!;

        public Candidatura Candidatura { get; private set; } = null!;

        protected MotivoCassacao() { }
    }
}
